servicetalk-concurrent-jdkflow/build.gradle

```diff
@@ -20,13 +20,12 @@ if (!JavaVersion.current().isJava9Compatible()) {
     project.tasks.all { task -> task.enabled = false }
 }
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_1_9
-    targetCompatibility = JavaVersion.VERSION_1_9
+compileJava {
+    options.release = 9
```

Comment:
Can we keep `sourceCompatibility` and `targetCompatibility` too? IIRC, some other tools, like IDE, look at them to generate project meta-data correctly.

Reply:
Sure, as long as it does not cause problems. My reading of https://docs.gradle.org/current/userguide/building_java_projects.html#sec:java_cross_compilation

was that these had been superseded by `options.release` and were no longer required.

Reply:
> Using a toolchain makes it illegal to configure the `sourceCompatibility` or `targetCompatibility` at the java { } extension level.

It appears that these properties are partially obsolete and may eventually be deprecated. I had looked at using `toolchain` rather than `options.release` but it doesn't quite do what we need.

Reply:
Until they are officially deprecated, let's keep them. It doesn't hurt to have some redundancy. Those options have 1-1 translation to pom.xml properties and may be useful for IDE.

Reply:
I am fine with keeping them as long as they don't cause problems.